src/lib/VirtualScroller.js

```diff
@@ -263,77 +263,70 @@ class VirtualScroller {
-            // then clone elements from curStartOffset to newEndOffset
-            this.cloneItems(newListEl, this.listEl, 0, newEndOffset - curStartOffset);
+            // Delete the elements from newEndOffset to the end
+            this.deleteItems(this.listEl, newEndOffset - curStartOffset + 1);
+            // Insert before the firstElementChild of the listEl
+            this.listEl.insertBefore(newListEl, this.listEl.firstElementChild);
         } else {
             // Scenario #3: New range has no overlap with current range of items
             //                          |--------------------|
             //                    curStartOffset        curEndOffset
             //  |--------------------|
             // newStartOffset    newEndOffset
             this.createItems(newListEl, newStartOffset, newEndOffset);
+            this.listEl.appendChild(newListEl);
         }
-
-        this.scrollingEl.replaceChild(newListEl, this.listEl);
-        this.listEl = newListEl;
     }
 
     /**
-     * Clones a subset of the HTMLElements from the oldList to the newList.
-     * The newList element is modified.
+     * Creates new HTMLElements appended to the newList
      *
      * @param {HTMLElement} newListEl - the new `ol` element
-     * @param {HTMLElement} oldListEl - the old `ol` element
      * @param {number} start - start index
      * @param {number} end  - end index
      * @return {void}
      */
-    cloneItems(newListEl, oldListEl, start, end) {
-        if (!newListEl || !oldListEl || start < 0 || end < 0) {
-            return;
-        }
-
-        const { children } = oldListEl;
-
-        if (!children || start >= children.length || end >= children.length) {
+    createItems(newListEl, start, end) {
+        if (!newListEl || start < 0 || end < 0) {
             return;
         }
 
         for (let i = start; i <= end; i++) {
-            newListEl.appendChild(children[i].cloneNode(true));
+            const newEl = this.renderItem(i);
+            newListEl.appendChild(newEl);
         }
     }
 
     /**
-     * Creates new HTMLElements appended to the newList
+     * Deletes elements of the 'ol'
      *
-     * @param {HTMLElement} newListEl - the new `li` element
+     * @param {HTMLElement} listEl - the `ol` element
      * @param {number} start - start index
-     * @param {number} end  - end index
+     * @param {number} [end] - end index
      * @return {void}
      */
-    createItems(newListEl, start, end) {
-        if (!newListEl || start < 0 || end < 0) {
+    deleteItems(listEl, start, end) {
+        if (!listEl || start < 0 || end < 0) {
             return;
         }
 
-        for (let i = start; i <= end; i++) {
-            const newEl = this.renderItem(i);
-            newListEl.appendChild(newEl);
-        }
+        const listItems = Array.prototype.slice.call(listEl.children, start, end);
+        listItems.forEach((listItem) => listEl.removeChild(listItem));
```

Comment:
Do we incur a performance penalty here, since `this.listEl` is currently mounted? Each `removeChild` call will force a reflow, right?